metaflow/plugins/events_decorator.py

```diff
@@ -632,14 +616,26 @@ def format_deploytime_value(self):
             old_trig = trigger
             if isinstance(trigger, DeployTimeField):
                 trigger = deploy_time_eval(trigger)
-            if isinstance(trigger, dict):
-                trigger["fq_name"] = trigger.get("name")
-                trigger["project"] = trigger.get("project")
-                trigger["branch"] = trigger.get("project_branch")
-            # We also added this bc it won't be formatted yet
-            if isinstance(trigger, str):
-                trigger = {"fq_name": trigger}
-                trigger = self._parse_fq_name(trigger)
+                if isinstance(trigger, dict):
+                    if "name" not in trigger:
+                        raise MetaflowException(
+                            f"Trigger: {trigger} parsed at deploy time is missing the *name* key."
+                        )
+                    if "project" not in trigger:
+                        raise MetaflowException(
+                            f"Trigger: {trigger} parsed at deploy time is missing the *project* key."
+                        )
+                    if "project_branch" not in trigger:
+                        raise MetaflowException(
+                            f"Trigger: {trigger} parsed at deploy time is missing the *project_branch* key."
+                        )
```

Comment:
this seems like a clear change in behaviour. previously partial dictionaries were accepted, though supplying a project without a branch lead to nothing being triggered. a dict with only a name used to be valid though

Reply:
Yes -- @talsperre , why are we forcing this now? It should be auto-filled with current project/branch if not provided right?